File: Exercism Codes/Kotlin/ScrabbleScore.kt
object ScrabbleScore {

    fun scoreLetter(c: Char): Int {
        return when(c) {
            'A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T' -> 1
            'D', 'G' -> 2
            'B', 'C', 'M', 'P' -> 3
            'F', 'H', 'V', 'W', 'Y' -> 4
            'K' -> 5
            'J', 'X' -> 8
            'Q', 'Z' -> 10
            else -> throw IllegalArgumentException()
        }
    }

    fun scoreWord(word: String): Int {
        var count : Int = 0
        for(i in word.uppercase()) count += scoreLetter(i)
        return count
    }
}

File: Exercism Codes/C#/ScrabbleScore.cs
public static class ScrabbleScore
{
    public static int Score(string input)
    {
        string word = input.ToLower();
        int score = 0;
        foreach (char c in word)
        {
            switch (c)
            {
                case 'a':
                case 'e':
                case 'i':
                case 'o':
                case 'u':
                case 'l':
                case 'n':
                case 'r':
                case 's':
                case 't':
                    score += 1;
                    break;
                case 'd':
                case 'g':
                    score += 2;
                    break;
                case 'b':
                case 'c':
                case 'm':
                case 'p':
                    score += 3;
                    break;
                case 'f':
                case 'h':
                case 'v':
                case 'w':
                case 'y':
                    score += 4;
                    break;
                case 'k':
                    score += 5;
                    break;
                case 'j':
                case 'x':
                    score += 8;
                    break;
                case 'q':
                case 'z':
                    score += 10;
                    break;
            }
        }
        return score;
    }
}